### PR TITLE
SUPP0RT-740: Handled filename issues

### PR DESCRIPTION
**Pull request by jekuaitk**
https://jira.itkdev.dk/browse/SUPP0RT-740

* Ensures no slash or backslashes in filename during view document action.

### CODE REVIEW

File: src/Service/DocumentUploader.php [hunk 1]
```diff
@@ -102,21 +102,22 @@ public function handleViewDocument(Document $document, bool $forceDownload = fal
             stream_copy_to_stream($fileStream, $outputStream);
         });
 
-        // We use document name as filename.
-        $filename = $document->getDocumentName();
+        // We use document name as filename - it may not contain / or \, so we replace those.
+        $documentName = $document->getDocumentName();
+        $ext = pathinfo($document->getFilename(), PATHINFO_EXTENSION);
```
Comment (rimi-itk) [inline src/Service/DocumentUploader.php:107]: Why do you rename `$filename` til `$documentName`? “filename” is a much better name for a variable that contains the name of a file …

Comment (rimi-itk) [inline src/Service/DocumentUploader.php:107]: Move this declaration to just before where it's used.

File: src/Service/DocumentUploader.php [hunk 2]
```diff
@@ -102,21 +102,22 @@ public function handleViewDocument(Document $document, bool $forceDownload = fal
...
 
-        $slugger = new AsciiSlugger();
-        $fallbackFilename = $slugger->slug($document->getDocumentName())->__toString();
+        $documentName = str_replace(['/', '\\'], '-', $documentName);
```
Comment (rimi-itk) [inline src/Service/DocumentUploader.php:109]: You may want to replace `%` as well since it's not allowed in the fallback filename.

**Review by rimi-itk — CHANGES_REQUESTED** [review]

File: src/Service/DocumentUploader.php (hunk 1 above)
Comment (jekuaitk) [reply src/Service/DocumentUploader.php:107]: Moved.

Comment (rimi-itk) [reply src/Service/DocumentUploader.php:107]: A variable name should not reflect the how the value has been found, but should reflect what it's value is.

File: src/Service/DocumentUploader.php (hunk 2 above)
Comment (jekuaitk) [reply src/Service/DocumentUploader.php]: Escaped `%`.

**Review by rimi-itk — APPROVED** [review]

**Pull request merged**